velox/functions/prestosql/aggregates/tests/PrestoHasherTest.cpp

```diff
@@ -0,0 +1,371 @@
+      PrestoHasher someHash(vector->type());
+      BufferPtr someHashes =
+          AlignedBuffer::allocate<int64_t>(someRows.end(), pool_.get());
+      someHash.hash(vector, someRows, someHashes);
+      auto someRawHashes = someHashes->as<int64_t>();
+      someRows.applyToSelected([&](auto row) {
+        EXPECT_EQ(expected[row], someRawHashes[row]) << "at " << row;
+      });
+    };
+
+    checkHashes(vector, expected);
+
+    // Wrap in Dictionary
+    BufferPtr indices =
+        AlignedBuffer::allocate<vector_size_t>(vectorSize, pool_.get());
+    auto rawIndices = indices->asMutable<vector_size_t>();
+    // Just 1:1 mapping.
+    for (size_t i = 0; i < vectorSize; ++i) {
+      rawIndices[i] = i;
+    }
+    auto dictionaryVector = BaseVector::wrapInDictionary(
+        BufferPtr(nullptr), indices, vectorSize, vector);
+
+    checkHashes(dictionaryVector, expected);
+
+    // Repeating rows.
+    std::vector<int64_t> modifiedExpected;
+    modifiedExpected.reserve(vectorSize);
+
+    for (size_t i = 0; i < vectorSize; ++i) {
+      rawIndices[i] = i / 2;
+      modifiedExpected[i] = expected[i / 2];
+    }
+
+    dictionaryVector = BaseVector::wrapInDictionary(
+        BufferPtr(nullptr), indices, vectorSize, vector);
+
+    checkHashes(dictionaryVector, modifiedExpected);
+
+    // Subset of rows.
+    auto subsetSize = vectorSize / 2;
+    BufferPtr subsetIndices =
+        AlignedBuffer::allocate<vector_size_t>(subsetSize, pool_.get());
+    auto rawSubsetIndices = subsetIndices->asMutable<vector_size_t>();
+    std::vector<int64_t> subsetExpected;
+    subsetExpected.reserve(vectorSize / 2);
+
+    for (size_t i = 0; i < vectorSize / 2; ++i) {
+      rawSubsetIndices[i] = i * 2;
+      subsetExpected[i] = expected[i * 2];
+    }
+
+    dictionaryVector = BaseVector::wrapInDictionary(
+        BufferPtr(nullptr), subsetIndices, subsetSize, vector);
+
+    checkHashes(dictionaryVector, subsetExpected);
+
+    // Wrap in a constant vector of various sizes.
+
+    for (int size = 1; size <= vectorSize - 1; size++) {
+      for (int i = 0; i < vectorSize - size; i++) {
+        auto constantVector = BaseVector::wrapInConstant(size, i, vector);
+        std::vector<int64_t> constantExpected(size, expected[i]);
+        checkHashes(constantVector, constantExpected);
+      }
+    }
+  }
+
+  template <typename T>
+  void testIntegral() {
+    assertHash<T>(
+        {1, 0, 100, std::nullopt},
+        {9155312661752487122, 0, -367469560765523433, 0});
+  }
+
+  VectorPtr createMapOfArraysVector(
+      std::map<
+          std::optional<int64_t>,
+          std::optional<std::vector<std::optional<int64_t>>>> map) {
+    std::vector<std::optional<int64_t>> keys;
+    std::vector<std::optional<std::vector<std::optional<int64_t>>>> values;
+    for (const auto& [key, vals] : map) {
+      keys.push_back(key);
+      values.push_back(vals);
+    }
+
+    auto arrayValues = vectorMaker_.arrayVectorNullable(values);
```

Comment:
Is there no corresponding makeNullableArrayVector in the base class? Let's add one.